### PR TITLE
Split list of emote sets into bunches when performing Ivr API reqeusts

### PR DESCRIPTION
**Pull request by zneix**
Pull request checklist:

- [x] `CHANGELOG.md` was updated, if applicable

# Description

Added a helper vector of vectors and some logic that splits QStringList that we receive as a function parameter to avoid getting 400's from Ivr API as it supports fetching 100 emote sets at max.

One unrelated change: updated a link to Ivr API endpoint we're using because it has changed.

P.S. Thanks @talneoran for helping to come up with the solution Okayga

### CODE REVIEW

File: src/providers/twitch/TwitchAccount.cpp [hunk 1]
```diff
@@ -301,54 +301,80 @@ void TwitchAccount::loadUserstateEmotes(QStringList emoteSetKeys)
...
+                    IvrEmoteSet ivrEmoteSet(emoteSet.toObject());
 
-                    emoteData->allEmoteNames.push_back(cleanCode);
+                    newUserEmoteSet->key = ivrEmoteSet.setId;
 
-                    auto twitchEmote =
-                        getApp()->emotes->twitch.getOrCreateEmote(id, code);
-                    emoteData->emotes.emplace(code, twitchEmote);
+                    auto name = ivrEmoteSet.login;
+                    name.detach();
```
Comment (Mm2PL) [inline src/providers/twitch/TwitchAccount.cpp:341]: Why is this needed?

File: src/providers/twitch/TwitchAccount.cpp [hunk 2]
```diff
@@ -301,54 +301,80 @@ void TwitchAccount::loadUserstateEmotes(QStringList emoteSetKeys)
...
-                newUserEmoteSet->type = QString();
-                newUserEmoteSet->channelName = ivrEmoteSet.login;
+    for (const auto &batch : std::move(batches))
+    {
+        QStringList emoteSetBatch;
+
+        for (const auto &el : batch)
+        {
+            emoteSetBatch.push_back(el);
+        }
```
Comment (Mm2PL) [inline src/providers/twitch/TwitchAccount.cpp:326]: If `batches` was `std::vector<QStringList>` (or something along those lines) from the start you wouldn't have to do this.

File: src/providers/twitch/TwitchAccount.cpp [hunk 3]
```diff
@@ -301,54 +301,80 @@ void TwitchAccount::loadUserstateEmotes(QStringList emoteSetKeys)
...
+                    for (const auto &emote : ivrEmoteSet.emotes)
+                    {
+                        IvrEmote ivrEmote(emote.toObject());
+
+                        auto id = EmoteId{ivrEmote.id};
+                        auto code = EmoteName{ivrEmote.code};
+                        auto cleanCode =
+                            EmoteName{TwitchEmotes::cleanUpEmoteCode(code)};
+                        newUserEmoteSet->emotes.emplace_back(
+                            TwitchEmote{id, cleanCode});
```
Comment (Mm2PL) [inline src/providers/twitch/TwitchAccount.cpp:357]: This constructs a `TwitchEmote` and then passes it to the copy constructor (via `emplace_back`), feels weird to me. I believe you either want
```suggestion
                        newUserEmoteSet->emotes.emplace_back(
                            id, cleanCode);
```

or

```suggestion
                        newUserEmoteSet->emotes.push_back(
                            TwitchEmote{id, cleanCode});
```

**Review by Mm2PL — CHANGES_REQUESTED** [review]
mostly comments :)

File: src/providers/twitch/TwitchAccount.cpp (hunk 1 above)
Comment (zneix) [reply src/providers/twitch/TwitchAccount.cpp:341]: Copy-pasta from `TwitchAccount::loadEmoteSetData`, it wasn't actually introduced here, but in #2550 - this is just indentation change - however if this should be changed I could as well make a change in this PR.

File: src/providers/twitch/TwitchAccount.cpp (hunk 2 above)
Comment (zneix) [reply src/providers/twitch/TwitchAccount.cpp:326]: Thank you!

Comment (zneix) [reply src/providers/twitch/TwitchAccount.cpp]: This works fine on Qt 5.15, however it doesn't work so well on Qt 5.12, so leaving this workaround for now.

**Review by Mm2PL — APPROVED** [review]

**Review by pajlada — APPROVED** [review]
